src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -1656,22 +1656,51 @@ private void AppendSymbolsWithName(ArrayBuilder<Symbol> results, string name, Bi
 
                     if (!wasError)
                     {
-                        results.Add(singleSymbol);
+                        results.Add(RemapSymbolIfNecessary(singleSymbol));
                     }
                     else
                     {
-                        results.AddRange(lookupResult.Symbols);
+                        foreach (var symbol in lookupResult.Symbols)
+                        {
+                            results.Add(RemapSymbolIfNecessary(symbol));
+                        }
                     }
                 }
                 else
                 {
-                    results.AddRange(lookupResult.Symbols);
+                    foreach (var symbol in lookupResult.Symbols)
+                    {
+                        results.Add(RemapSymbolIfNecessary(symbol));
+                    }
                 }
             }
 
             lookupResult.Free();
         }
 
+        private Symbol RemapSymbolIfNecessary(Symbol symbol)
+        {
+            if (!Compilation.NullableSemanticAnalysisEnabled) return symbol;
+
+            switch (symbol)
+            {
+                case LocalSymbol _:
+                case ParameterSymbol _:
+                case MethodSymbol { MethodKind: MethodKind.LambdaMethod }:
+                case MethodSymbol { MethodKind: MethodKind.LocalFunction }:
+                    return RemapSymbolIfNecessaryCore(symbol, symbol.Locations[0].SourceSpan.Start);
```

Comment:
>Locations[0] [](start = 69, length = 12)

Consider making code more robust by checking that we actually have a location. Perhaps asserting and skipping symbols that do not have locations.